### PR TITLE
Docs: Add selectableTypes example to ObjectBrowserWidget storybook

### LINKED ISSUE #2993: More options for the object browser's selectable content types
The object browser currently only has one way to filter the selectable content types is to set the mode to 'image', which forces you to select an image. The object browser's image mode currently works by checking `settings.imageObjects` for a list of allowed image objects. It would be useful if it were possible to apply similar restrictions to other content types, such as for use in the video block or for a file picker.

## Options
- Have various modes to support a number of various out-the-box modes.
  - This would not be useful for custom content-types unless the list of modes was adjustable, such as through a config setting.
- Have the user pass in the short name of allowed content-types to `openObjectBrowser`.

### PR DESCRIPTION
**Pull request by JeffersonBledsoe**
Just a quick PR to add an example of the `selectableTypes` to the `ObjectBrowserWidget`'s storybook.

Closes #2993

### CODE REVIEW

File: docs/source/blocks/editcomponent.md
Comment (stevepiercy) [inline docs/source/blocks/editcomponent.md:293]: I think that "selectableTypes" is an inline literal, but if not then please ignore this suggestion. This content is good. Thank you!

```suggestion
If `selectableTypes` is set in `widgetOptions.pattern_options`, then only items whose content type has a name that is defined in `widgetOptions.pattern_options.selectableTypes` will be selectable.
```
```suggestion
If **selectableTypes** is set in `widgetOptions.pattern_options`, then only items whose content type has a name that is defined in `widgetOptions.pattern_options.selectableTypes` will be selectable.
```

**Review by stevepiercy — APPROVED** [review]
LGTM. Let's get a core maintainer to review as well before merging.

**Review by sneridagh — APPROVED** [review]

**Comment by sneridagh** [comment]
@JeffersonBledsoe awesome! thanks!

**Pull request merged**
